### PR TITLE
feat: add install script to improve talosctl installation 

### PR DESCRIPTION
**Pull request by steverfrancis**
This install script detects the platform and architecture, and downloads the correct talosctl, and checks the gpg checksums. It also installs and chmods the binary.

Signed-off-by: Steve Francis <steve.francis@talos-systems.com>

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

File: website/static/install [hunk 1]
```diff
@@ -0,0 +1,135 @@
+#!/bin/sh
```
Comment (andrewrynhard) [inline website/static/install:1]: `#!/usr/bin/env sh` could offer broader compatibility.

**Review by andrewrynhard — APPROVED** [review]
Minor nit.

File: website/static/install [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+set -eu
+
+INSTALLPATH=/usr/local/bin
```
Comment (frezbo) [inline website/static/install:5]: I would make this user settable: 

for eg, many folks might not `~/.bin` or `~/.local/bin`: eg

`curl -SL https://raw.githubusercontent.com/siderolabs/talos/6f99922f681522066bc6aaae40efdd9ec728cc2d/website/static/install | INSTALLPATH=/tmp sh`

```suggestion
INSTALLPATH=${INSTALLPATH:-/usr/local/bin}
```

File: website/static/install [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+happyexit() {
+  echo "Installed version:"
+  echo "$(talosctl version 2>/dev/null )"
```
Comment (frezbo) [inline website/static/install:9]: same here, use the full path:

```suggestion
  echo "$(${INSTALLPATH}/talosctl version 2>/dev/null )"
```

Comment (frezbo) [reply website/static/install:9]: that's a good point, for me when I tested the modifed version of the script with `/tmp` it actually picked up my normal talosctl install, if the user specifies a custom path I guess it's upto them to make sure it's in their search path I guess, or maybe we could just check if there's already a talosctl in normal path and warn them of a custom path

Comment (steverfrancis) [reply website/static/install:9]: I think that if people are setting INSTALLPATH, they know about search paths, and would want to see the version that is called first... ANd can adjust path if necessary.
So I'd rather say , as you said " if the user specifies a custom path I guess it's upto them to make sure it's in their search path "

File: website/static/install [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+OS=$(uname -s)
+arch=$(uname -m)
+cli_arch=""
+case $OS in
+  CYGWIN* | MINGW64*)
+    OS=windows-amd64.exe
+    ;;
+  Darwin | Linux | FreeBSD)
```
Comment (smira) [inline website/static/install:52]: I don't remember how WSL 2 reports itself, but it'd be nice to have it detected as Linux

File: website/static/install (hunk 1 above)
Comment (steverfrancis) [inline website/static/install:1]:
```suggestion
#!/usr/bin/env sh```

**Review by steverfrancis — COMMENTED** [review]
Implemented @andrewrynhard 's suggestion

**Review by smira — APPROVED** [review]

**Pull request merged**